### PR TITLE
Split config into multiple bitly requests

### PR DESCRIPTION
**Pull request by aminroosta**

### CODE REVIEW

File: src/store/ChartStore.js [hunk 1]
```diff
@@ -218,38 +218,43 @@ class ChartStore {
...
-                    layoutData = decodeURI(sharedParams);
-                    try {
-                        layoutData = JSON.parse(layoutData);
-                    }catch(e){
-                        console.error(e);
-                    }
-                    window.history.replaceState({}, document.title, window.location.pathname);
-                }
-            }
+            const doTheRest = () => {
```
Comment (brucebinary) [inline src/store/ChartStore.js]: this needs a better name

File: src/store/ChartStore.js [hunk 2]
```diff
@@ -218,38 +218,43 @@ class ChartStore {
...
-            } else if (stxx.chart.symbol) {
-                this.currentActiveSymbol = stxx.chart.symbolObject;
-                stxx.chart.yAxis.decimalPlaces = stxx.chart.symbolObject.decimal_places;
-                this.categorizedSymbols = this.categorizeActiveSymbols();
+                this.context = context;
+                this.contextPromise.resolve(this.context );
+                this.resizeScreen();
+                this.chartPanelTop = holderStyle.top;
+            };
+            const configParams = window.location.href.split('#');
```
Comment (brucebinary) [inline src/store/ChartStore.js]: this line is not used anywhere

File: src/store/ShareStore.js [hunk 3]
```diff
@@ -29,52 +29,100 @@ export default class ShareStore {
...
+    }
+    expandBitlyAsync(hash, payload) {
+        if(hash === 'NONE') {
+            return Promise.resolve(payload);
+        }
+        const bitlyLink = `${window.location.protocol}//bit.ly/${hash}`;
+        return fetch(`${this.bitlyUrl}/expand?access_token=${this.accessToken}&shortUrl=${bitlyLink}`)
+            .then(response => response.json())
+            .then(response =>  {
+                if (response.status_code == 200 ) {
```
Comment (brucebinary) [inline src/store/ShareStore.js]: use `===`. Same for other parts of the PR

File: src/store/ShareStore.js [hunk 4]
```diff
@@ -29,52 +29,100 @@ export default class ShareStore {
...
             .catch(error => {
                 self.loading = false;
                 self.urlGenerated = false;
             });
+    }
+    shortenBitlyAsync(payload, hash) {
+        let origin = window.location.href;
+        origin = origin.replace('localhost', '127.0.0.1'); // make it work on localhost
+
+        //const shareLink = this.fixedEncodeURIComponent(`${origin}?${hash}#${payload}`);
```
Comment (brucebinary) [inline src/store/ShareStore.js]: unused code

File: src/store/ShareStore.js [hunk 5]
```diff
@@ -29,52 +29,100 @@ export default class ShareStore {
...
+        return fetch(`${this.bitlyUrl}/shorten?access_token=${this.accessToken}&longUrl=${shareLink}`)
+            .then(response => response.json())
+            .then(response =>  {
+                if (response.status_code == 200 ) {
+                    return response.data.url.split('bit.ly/')[1];
+                }
+                return null;
+            });
+    }
+    expandBitlyAsync(hash, payload) {
```
Comment (brucebinary) [inline src/store/ShareStore.js]: just a thought: this function is only used in `ChartStore.js`. Though... It seems more appropriate to place it in `ShareStore.js` even though it is not used there.

File: src/store/ShareStore.js [hunk 6]
```diff
@@ -29,52 +29,100 @@ export default class ShareStore {
...
-            fixedEncodeURIComponent = function (str) {
-                return encodeURIComponent(str).replace(/[!'()*]/g, function(c) {
-                    return `%${c.charCodeAt(0).toString(16)}`;
-                });
-            };
+        const json = JSON.stringify(layoutData);
 
         const origin = window.location.href;
-        this.shareLink = fixedEncodeURIComponent(`${origin}#${json}`);
+        const encodedJson = encodeURIComponent(json); // this.fixedEncodeURIComponent(json);
```
Comment (brucebinary) [inline src/store/ShareStore.js]: What's this `// this.fixedEncodeURIComponent(json);`?

**Comment by aminroosta** [comment]
@brucebinary fixed issues.

I hade a merge conflict like this, that i resolved to HEAD

screen shot 2018-05-22 at 9 48 33 am

Make sure to bring back that  `checkForQueryStringPart` if you need to.

**Comment by brucebinary** [comment]
@aminroosta I'm aware of that `checkForQueryStringPart`. I deliberately had it ignored because we might want to check this feature in github pages as well.

...wait. The shared URL link should always point to charts.binary.com. This feature will break when using with binary static.

**Pull request merged**